src/features/profile/pages/ChangeEmail/ChangeEmail.native.test.tsx

```diff
@@ -47,151 +27,233 @@ jest.mock('ui/components/snackBar/SnackBarContext', () => ({
   })
 
-  it('should render and log correctly when an email change is already in progress', async () => {
-    mockedUseQuery.mockImplementationOnce(() => ({
-      data: { expiration: '2021-12-07T13:45:05.812190' },
-    }))
-    const renderAPI = render(<ChangeEmail />)
-    expect(renderAPI.toJSON()).toMatchSnapshot()
+  describe('email change already in progress', () => {
+    it('should render correctly', async () => {
+      simulateCurrentEmailChange()
+      const renderAPI = renderChangeEmail()
 
-    await waitForExpect(() => {
-      expect(analytics.logConsultDisclaimerValidationMail).toHaveBeenCalledTimes(1)
+      await superFlushWithAct()
+      expect(renderAPI).toMatchSnapshot()
+    })
+
+    it('should log analytics', async () => {
+      simulateCurrentEmailChange()
+      renderChangeEmail()
+
+      await waitFor(() => {
+        expect(analytics.logConsultDisclaimerValidationMail).toHaveBeenCalledTimes(1)
+      })
     })
   })
 
-  it.each`
-    password          | email                | isDisabled
-    ${'password>=12'} | ${'valid@email.com'} | ${false}
-    ${'password>=12'} | ${'invalid@email'}   | ${true}
-    ${'password<12'}  | ${'valid@email.com'} | ${true}
-    ${'password>=12'} | ${''}                | ${true}
-  `(
-    'CTA "Enregistrer les modifications" (disabled=$isDisabled) with background color = $backgroundColor if password = "$password" and email = $email',
-    async ({ password, email, isDisabled }) => {
-      const { getByPlaceholderText, getByLabelText } = render(<ChangeEmail />)
+  describe('submit button', () => {
+    it('should be disabled by default', () => {
+      const { getByLabelText } = renderChangeEmail()
       const submitButton = getByLabelText('Enregistrer les modifications')
+
       expect(submitButton).toBeDisabled()
+    })
 
-      await act(async () => {
-        const passwordInput = getByPlaceholderText('Ton mot de passe')
-        fireEvent.changeText(passwordInput, password)
-      })
-      await act(async () => {
-        const emailInput = getByPlaceholderText('tonadresse@email.com')
-        fireEvent.changeText(emailInput, email)
+    it.each`
+      password          | email                | isDisabled
+      ${'password>=12'} | ${'valid@email.com'} | ${false}
+      ${'password>=12'} | ${'invalid@email'}   | ${true}
+      ${'password<12'}  | ${'valid@email.com'} | ${true}
+      ${'password>=12'} | ${''}                | ${true}
+    `(
+      'should have state (disabled=$isDisabled) with background color = $backgroundColor when password = "$password" and email = $email',
+      async ({ password, email, isDisabled }) => {
+        const { getByPlaceholderText, getByLabelText } = renderChangeEmail()
+        const submitButton = getByLabelText('Enregistrer les modifications')
+
+        await fillInputs(getByPlaceholderText, { email, password })
+
+        expect(submitButton)[isDisabled ? 'toBeDisabled' : 'toBeEnabled']()
+      }
+    )
+  })
```

Comment:
Ce it.each a plus de sens : on test la validation avec le schema déjà, on doit juste tester l'intégration, donc deux tests séparés sont suffisant (et permet d'enlever le paramètrage `[isDisabled ? 'toBeDisabled' : 'toBeEnabled']()` qui est pas terrible (cf https://github.com/pass-culture/pass-culture-app-native/pull/4115#discussion_r1057196068)

Tu peux donc juste rajouter deux tests : 
- should disabled submit button when form is not valid
- should enabled submit button when form is valid

(ça permet aussi de retirer la mention du background color qui est même pas dans les params du each...)